microsoft/testsuites/vm_extensions/azurediskencryption.py

```diff
@@ -0,0 +1,200 @@
+
+    node_capability = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+    location = node_capability.location
+    shared_resource_group = runbook.shared_resource_group_name
+
+    # Create key vault if there is not available adelisa key vault for that region
+    existing_vault = get_matching_key_vault_name(
+        platform, location, shared_resource_group, r"adelisa-\w{5}"
+    )
+    if existing_vault:
+        vault_name = existing_vault
+    else:
+        random_str = generate_random_chars(string.ascii_lowercase + string.digits, 5)
+        vault_name = f"adelisa-{random_str}"
+
+    vault_properties = VaultProperties(
+        tenant_id=tenant_id,
+        sku=KeyVaultSku(name="standard"),
+        enabled_for_disk_encryption=True,
+        access_policies=[
+            AccessPolicyEntry(
+                tenant_id=tenant_id,
+                object_id=object_id,
+                permissions=Permissions(
+                    keys=["all"], secrets=["all"], certificates=["all"]
+                ),
+            ),
+        ],
+    )
+    # If the KV exists, this will just ensure the properties are correct for ADE
+    keyvault_result = create_keyvault(
+        platform=platform,
+        location=location,
+        vault_name=vault_name,
+        resource_group_name=shared_resource_group,
+        vault_properties=vault_properties,
+    )
+
+    # Check if KeyVault is successfully created or updated before proceeding
+    assert (
+        keyvault_result
+    ), f"Failed to create or update KeyVault with name: {vault_name}"
+
+    # Run ADE Extension
+    extension_name = "AzureDiskEncryptionForLinux"
+    extension_publisher = "Microsoft.Azure.Security"
+    extension_version = "1.4"
+    settings = {
+        "EncryptionOperation": "EnableEncryption",
+        "KeyVaultURL": keyvault_result.properties.vault_uri,
+        "KeyVaultResourceId": keyvault_result.id,
+        "KeyEncryptionAlgorithm": "RSA-OAEP",
+        "VolumeType": "Os",
+    }
+    extension = node.features[AzureExtension]
+    extension_result = extension.create_or_update(
+        name=extension_name,
+        publisher=extension_publisher,
+        type_=extension_name,
+        type_handler_version=extension_version,
+        settings=settings,
+    )
+    return extension_result
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="Tests for the Azure Disk Encryption (ADE) extension",
+    requirement=simple_requirement(
+        supported_features=[AzureExtension],
+        supported_platform_type=[AZURE],
+        supported_os=[Ubuntu, CBLMariner, CentOs, Oracle, Redhat, SLES, Suse],
+    ),
+)
+class AzureDiskEncryption(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        Runs the ADE extension and verifies it
+        fully encrypted the remote machine successfully.
+        """,
+        priority=3,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
```

Comment:
I encounter this error
 ```raise Exception("Not enough memory for enabling encryption on OS volume. 8 GB memory is recommended.")```, 

please add 8 Gb memory as case requirement
``` 
       requirement=node_requirement(
            supported_features=[AzureExtension],
            node=schema.NodeSpace(
                core_count=1, memory_mb=search_space.IntRange(min=8 * 1024)
            ),
        ),
```

Reply:
I added the node_requirement in TestSuiteMetadata rather than TestCase.

Reply:
The test case level requirement overwrites the test suite level requirement.

Reply:
@squirrelsc I was having an issue adding supported_features to node_requirement. It seemed to complain in mypy, and only work well with simple_requirement. Whats the best way to do this?

Reply:
Can you share the error met?

Reply:
mypy ./microsoft/testsuites/vm_extensions/azure_disk_encryption.py 
microsoft/testsuites/vm_extensions/azure_disk_encryption.py:134:21: error: Unexpected keyword argument "supported_features" for "node_requirement"
Found 1 error in 1 file (checked 1 source file)

Reply:
Also, if test case level overwrites, seems I should only use testcase level to define the metadata, since each test has a different priority.

Reply:
priority is not part of the requirements, so it doesn't affect the requirements by overwriting. The type should be node_requirement, instead of simple_requirement. Because the simple_requirement doesn't have the `supported_features` argument.

Reply:
I pasted the code I am using when seeing the error. The pushed code is not seeing the error. 

I see the error using node_requirement. simple_requirement does have the supported_features argument. I could remove the argument in general for node_requirement?

Reply:
Sorry, I mixed them incorrectly. It needs to use `simple_requirement` or use the `node_requirement` within a node space. BTW, I just found it set the core_count to 1, it means use a VM size with only 1 core. It would be very slow, make sure it's the test purpose.

```python
node_requirement(
  node=schema.NodeSpace(
    supported_features=[AzureExtension],
    memory_mb=search_space.IntRange(min=8 * 1024)
  )
)
```

Reply:
Yes, you can define core_count like memory_mb to set a min value.

Reply:
` @TestCaseMetadata(
        description="""
        Runs the ADE extension and verifies it
        fully encrypted the remote machine successfully.
        """,
        priority=3,
        timeout=3600*2,
        requirement=node_requirement(
            node=schema.NodeSpace(
                supported_features=[AzureExtension],
                memory_mb=search_space.IntRange(min=8 * 1024),
                core_count = 2
            ),
            supported_platform_type=[AZURE],
            supported_os=[Ubuntu, CBLMariner, CentOs, Oracle, Redhat, SLES, Suse],
        )
    )`


and 

`@TestCaseMetadata(
        description="""
        Runs the ADE extension and verifies the extension
        provisioned successfully on the remote machine.
        """,
        priority=1,
        requirement=node_requirement(
            node=schema.NodeSpace(
                memory_mb=search_space.IntRange(min=8 * 1024),
            ),
            supported_features=[AzureExtension],
            supported_platform_type=[AZURE],
            supported_os=[Ubuntu, CBLMariner, CentOs, Oracle, Redhat, SLES, Suse],
        )
    )`

are both still throwing :

microsoft/testsuites/vm_extensions/azure_disk_encryption.py:130:18: error: Unexpected keyword argument "supported_features" for "NodeSpace"
microsoft/testsuites/vm_extensions/azure_disk_encryption.py:197:21: error: Unexpected keyword argument "supported_features" for "node_requirement"

Reply:
Sorry, please try `features`, instead of `supported_features`. I found it's not used in test cases, so it may need to refer this example. @LiliDeng can you support min memory in simple requirements? So it doesn't need to use node_requirement in this case.

Reply:
Also having same troubles with features inside node_requirement.

I think I can add an optional parameter to node_requirement, which would then pass to _create_test_case_requirement, would you like me to push that change for you to see?

Reply:
Pushed the change that could work. I feel limited here though. For example, would be nice to also add a test using Data disks, which seems possible via simple_requirements. However, I would still want the node memory and core constraints.

Reply:
Please add the `min_memory_mb` supports to simple_requirement, instead of node. The features are part of node, not at the node_requirement level.

Reply:
> Pushed the change that could work. I feel limited here though. For example, would be nice to also add a test using Data disks, which seems possible via simple_requirements. However, I would still want the node memory and core constraints.

The `node_requirement` is more capable than simple_requirement, but it's complex. If you add the min_memory_mb to `simple_requirement`, the simple_requirement fits the requirements. But the node_requirement cannot add supported_features, because it should be in the node, not separatedly.